runtime/src/bank.rs

```diff
@@ -1210,16 +1211,55 @@ impl Bank {
         }
     }
 
+    fn distribute_rent_to_validators(
+        &self,
+        vote_account_hashmap: &HashMap<Pubkey, (u64, Account)>,
+        rent_to_be_distributed: u64,
+    ) {
+        let mut total_staked = 0;
+
+        let node_stakes = vote_account_hashmap
+            .iter()
+            .filter_map(|(_vote_pubkey, (staked, account))| {
+                total_staked += *staked;
+                VoteState::deserialize(&account.data)
+                    .ok()
+                    .map(|vote_state| (vote_state.node_pubkey, *staked))
+                    .filter(|(_pubkey, staked)| *staked != 0)
+            })
+            .collect::<HashMap<Pubkey, u64>>();
+
+        node_stakes.iter().for_each(|(pubkey, staked)| {
+            let rent_to_be_paid =
+                (((staked * rent_to_be_distributed) as f64) / (total_staked as f64)) as u64;
+            let mut account = self.get_account(pubkey).unwrap_or_default();
+            account.lamports += rent_to_be_paid;
+            self.store_account(pubkey, &account);
+        });
+    }
+
     fn distribute_rent(&self) {
         let total_rent_collected = self.collected_rent.load(Ordering::Relaxed);
 
-        if total_rent_collected != 0 {
-            let burned_portion =
-                (total_rent_collected * u64::from(self.rent_collector.rent.burn_percent)) / 100;
-            let _rent_to_be_distributed = total_rent_collected - burned_portion;
-            // TODO: distribute remaining rent amount to validators
-            // self.capitalization.fetch_sub(burned_portion, Ordering::Relaxed);
+        let (burned_portion, rent_to_be_distributed) = self
+            .rent_collector
+            .rent
+            .calculate_burn(total_rent_collected);
+
+        self.capitalization
+            .fetch_sub(burned_portion, Ordering::Relaxed);
+
+        if rent_to_be_distributed == 0 {
+            return;
+        }
+
+        let vote_account_hashmap = self.epoch_vote_accounts(self.epoch);
```

Comment:
I think you should use live stakes.  epoch_vote_accounts() is really just for leader schedules.

Reply:
@rob-solana yeah, it makes sense to use live stakes, as account credit is specific to this bank only.